### PR TITLE
Port Frontend to React

### LINKED ISSUE #1: Landingpage
Create the base landingpage in ./frontend/index.html.

### LINKED ISSUE #11: Minify CSS, HTML and JS before merging with Main branch
Remember to minify the frontend parts into a optimized branch before merging into the main branch for slight performance gain!

### LINKED ISSUE #31: Redesign cards
The card design hasn't been modified for the new card images, so they look a bit out of place. Tweaking the card design would make the site look a lot more interesting.

### LINKED ISSUE #53: Port to React
I'm gonna look into changing the site to use React and possibly Vite.

### PR DESCRIPTION
**Pull request by OscarLarm**
### Description
The frontend part of the project has been remade using React JS. This should make it easier to use libraries and to quickly add features. 
The react components being seperated makes everthing easier to read. It also makes it very simple if we want to reuse any component, or remove/change one specific component. The react component library Material UI has been added to the project, so the searchbar is now a Material UI component. There are several additional components which should be added to the project in the future. 
The project now also uses Vite to build/host the site instead of the launch_site.py script. Vite seems to be working well but if we'd like something else further down the road then it shouldn't be a problem to switch. The command to start the site through vite is npm run dev. Add -- --host if you want to expose addresses.
This also sort of fixes the issue of minifying the code, since vite has another command 'npm run build' to create a dist folder with minified files. I haven't set it up beyond the standard config but it should work if we want to host it somewhere.

### Details
- Port frontend to React JS
- Add Vite to project
- The project now uses the React component library Material UI
- The Searchbar is now a Material UI component
- Slight redesign to cards

### References
Close #53 
Close #11 
Close #31 

<br>

### CODE REVIEW

File: src/services/DataHandler.jsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+// Create a context to provide the data
+const DataContext = createContext();
+
+export function useData() {
+    return useContext(DataContext);
+}
+
+async function fetchKeywordInfoData() {
+  const url = 'http://192.168.0.112:5080/keyword-abilities';
```
Comment (annk15) [inline src/services/DataHandler.jsx]: We don't want to hard code addresses. Could we modify this so the address and port are fetched from an environment variables in a .env file like how it is done [here](https://create-react-app.dev/docs/adding-custom-environment-variables/#expanding-environment-variables-in-env)

File: src/services/DataHandler.jsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      })
+    );
+  } catch (error) {
+    console.error('Error fetching card data:', error);
+    return [];
+  }
+}
+
+async function fetchKeywordImageData(keyword) {
+  const url = `http://192.168.0.112:5080/keyword-image/${keyword}`;
```
Comment (annk15) [inline src/services/DataHandler.jsx]: Same as [here](https://github.com/annk15/MTGKeywordHelper/pull/56#discussion_r1729778899)

**Review by annk15 — CHANGES_REQUESTED** [review]
This looks really good! 🎉
I just want one small change before we merge

File: src/services/DataHandler.jsx (hunk 1 above)
Comment (OscarLarm) [reply src/services/DataHandler.jsx]: Made an enviroment file for the API addresses. We should probably add everything for flask, the DB and server as well. I've added an env.example as a preset. I'm gonna look into adding stuff to our README later with some instructions and examples for the .env values. Could be useful if we move all of /database/ and /server/ into .env. :smiley:

**Comment by OscarLarm** [comment]
I think I've added what you were looking for, rerequesting review! :smiley:

File: src/services/DataHandler.jsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      })
+    );
+  } catch (error) {
+    console.error('Error fetching card data:', error);
+    return [];
+  }
+}
+
+async function fetchKeywordImageData(keyword) {
+  const url = `${API_BASE_URL}/${API_IMAGE_URL}/${keyword}`;
```
Comment (annk15) [inline src/services/DataHandler.jsx]: I don't think we need to map the route as that wont be something that differs between environments.
So we can replace API_IMAGE_URL with the hard coded path and the same for API_KEYWORD_URL.
We could add a new variable for the ports however as the user may not be able to run on the defined ports in all cases.

So we could do just
const url = `${API_URL}/XXX/......`;
Or 
const url = `${API_URL}:${API_PORT}/XXX/......`;

**Review by annk15 — CHANGES_REQUESTED** [review]

File: src/services/DataHandler.jsx (hunk 3 above)
Comment (OscarLarm) [reply src/services/DataHandler.jsx]: :+1:

**Review by annk15 — COMMENTED** [review]
This is lovely! ❤️

**Review by annk15 — APPROVED** [review]

**Pull request merged**